# Inertial parameters referred to joints' frame(!) of DH parametrization to use regressor.
# Obtained with transformation of URDF file from:
#    - $(find franka_gazebo)/test/launch/panda-gazebo.urdf
#    - $(find franka_description)/robots/common/franka_robot.xacro
#    - $(find franka_description)/robots/common/inertial.yaml 
# Considering Denavit-Hartenberg parametrization of table:
# DH_table << 0,      -M_PI_2,  0.3330, 0,
#             0,      M_PI_2,   0,      0,
#             0.0825, M_PI_2,   0.3160, 0,
#             -0.0825,-M_PI_2,  0,      0,
#             0,      M_PI_2,   0.384,  0,
#             0.088,  M_PI_2,   0,      0,
#             0,      0,        0.107,  0;

link1_params: &link1
  mass: 4.8966058715389185
  m_CoM_x: 0.01898841848489417
  m_CoM_y: 0.22366182741323895
  m_CoM_z: 0.010347926586395934
  Ixx: 0.7206019233616886
  Ixy: -0.0072478112281814293
  Ixz: -0.00017399530451226307
  Iyy: 0.0080797264704072155
  Iyz: -0.020509431991783057
  Izz: 0.69335733649421039

link2_params: &link2
  mass: 0.56430313315832648
  m_CoM_x: -0.0020281277516575858
  m_CoM_y: 0.002303334975923871
  m_CoM_z: 0.017151233988577208
  Ixx: 0.008542099451553653
  Ixy: 0.010979338775564108
  Ixz: 0.0040697491272901499
  Iyy: 0.027023226541889683
  Iyz: -0.00075588448349189665
  Izz: 0.029913196027103954

link3_params: &link3
  mass: 3.1543494791973998
  m_CoM_x: -0.1850412686319525
  m_CoM_y: -0.21329271033496935
  m_CoM_z: -0.13009756542299949
  Ixx: 0.058094156813591431
  Ixy: -0.024733042974971586
  Ixz: -0.0022593428931625024
  Iyy: 0.026485842805219355
  Iyz: 0.0041846488665644018
  Izz: 0.063247127383446192

link4_params: &link4
  mass: 3.8378221280471703
  m_CoM_x: 0.098410380163609062
  m_CoM_y: -0.10489257893617226
  m_CoM_z: 0.3467410335712382
  Ixx: 0.070154616671045791
  Ixy: 0.004354917897079486
  Ixz: -0.0032740170750251108
  Iyy: 0.06897198164453848
  Iyz: 0.0016470034837128716
  Izz: 0.027027880696582539

link5_params: &link5
  mass: 1.219386216603884
  m_CoM_x: -0.013558365303777712
  m_CoM_y: -0.046288376973957476
  m_CoM_z: -0.046080574309319264
  Ixx: 0.038619624267928858
  Ixy: -0.0045018033035872373
  Ixz: 0.0013737972995086205
  Iyy: 0.01145854379145733
  Iyz: -0.0022283032261615314
  Izz: 0.029580002025284071

link6_params: &link6
  mass: 1.5410869099331015
  m_CoM_x: -0.04432935560820235
  m_CoM_y: -0.018305523892900209
  m_CoM_z: 0.022825237976701693
  Ixx: 0.0026099030386792046
  Ixy: -0.0017548535368365223
  Ixz: 0.00051705040384607563
  Iyy: 0.0072977991696814319
  Iyz: -9.0653082761660265e-05
  Izz: 0.0059521963230544017

link7_params: &link7
  mass: 0.76035445418479031
  m_CoM_x: 0.0081056735490182188
  m_CoM_y: -0.0033048625358797394
  m_CoM_z: -0.035973277768219125
  Ixx: 0.014246666049242025
  Ixy: -0.00042026977379739581
  Ixz: -0.0008745189275856047
  Iyy: 0.011371520950656969
  Iyz: -0.0007554720176194469
  Izz: 0.004638742862331415

default_controller: &default
  link1_REG: *link1
  link2_REG: *link2
  link3_REG: *link3
  link4_REG: *link4
  link5_REG: *link5
  link6_REG: *link6
  link7_REG: *link7

backstepping_controller: *default
